### ISSUE TITLE
Profainity Toggling

### ISSUE THREAD

**Issue by Hockeyfan360** [TotalFreedom/TotalFreedomMod#717]
Well, kind of the title. If a player wishes, they can use /toggle profainity, which would censor commonly used curse words, and replace them with stars. This, imo, should be toggled off by default.

**Comment by PattyToTheWatty** [comment]
Yes! My friend got in Trouble from he's parents because if other people swearing. Please implement it!

**Comment by JeromSar** [comment]
@Wild1145 What do you think?

**Comment by Wild1145** [comment]
@JeromSar I think as long as its filtered player by player (so not everyone) it should be quite good. I know there are a fair few players that do not appreciate the profanity on the server, so having the ability for it to star those words out or similar would be a good addition.

**Comment by Hockeyfan360** [comment]
@Wild1145 @JeromSar

Only problem is, how are we going to add these words in? Hardcode? Config? - If we did it in config, those who don't prefer profainity would be forced to see this. People working with tfm would be forced to see them if it was hard-coded. However, implementation other than toggling should be fairly easy.

**Comment by Wild1145** [comment]
@Hockeyfan360 @JeromSar I'd have it all in a config, and then have a list of players that toggled not to see the words stored in memory, and on a chat event if a chat is sent with any of those words in it, the word is replaced by stars and then sent, if not it just continued.

**Comment by Robolawrence** [comment]
Sounds like a good idea!

**Comment by Hockeyfan360** [comment]
@Wild1145 Memory as in a list getting wiped every time the server stops? (been a while since i used tfm, so if the files don't exist, well, sorry) If there are playerdata files for a player, it could be a boolean, so you don't have to change it often.

**Comment by Wild1145** [comment]
@Hockeyfan360 Oh yeah, I forgot we have player data files, or at least I think we still do (Its been a long time myself since I played with TFM)... @JeromSar would have to confirm, but if we still do have player data files, you could just opt in or out of not seeing profanity...

**Comment by tylerhyperHD** [comment]
I think the child play here act may make this a priority as any public sexual language is considered a violation.

**Comment by Hockeyfan360** [comment]
@tylerhyperHD I believe that the act only applies to libraries and schools that are funded by the government. In other words, they have to censor stuff like that. It is none of our concern.

**Comment by Wild1145** [comment]
We dont have to filter anything, its down to ISP's or the computer owners to take such action. Not us.

**Comment by JeromSar** [comment]
After some consideration, I've decided to accept this issue, but I most likely will not be working on it for now. Feel free to make PRs. :)
